### DISCUSSION TITLE
WriteBuffer performance Issue

### DISCUSSION THREAD

**Discussion by lslzl3000** [gpuweb/gpuweb#2600] [Q&A] [answered]
I have noticed a very common problem on `writebuffer` performance:

1. With intel hd graphics, same computer, Windows will gain over 20% performance compared to Mac os. Why? Is there any difference on system level? Or Dawn has different approach on different system?

2. Compared to Webgl, `writebuffer ` is like `gl.bufferSubData`, but it comes very bed performance when write a large data. 
On my demo test, e,g, submit a 300k transform matrix buffer ( 300k * 16 = 4,800,000 array) to gpu side
The `gl.bufferSubData` will only take about 3~4 ms,  where `writebuffer` could take over 50ms, up to 70ms sometime, very slowly and unstable,  any idea why?

***
Chrome canary v100+
Mac os 12, Windows 10
Intel UHD 630
Intel i7 8700k
16gb DDR4
***

**Comment by lslzl3000** [comment]
By the way, the `map/unmap` way takes event more time then `writebuffer` in this case.

**Comment by kvark** [comment, marked as answer]
Please ask on Chromium places specifically. I don't believe there is anything in the specification or the API that would make writeBuffer slow.
See https://github.com/gpuweb/gpuweb/issues/2601#issuecomment-1040155926
